javascript/javascript2/week2/products/main.js

```diff
@@ -0,0 +1,98 @@
+console.log("Script loaded");
+
+const products = getAvailableProducts();
+const productsUl = document.querySelector("section.products ul");
+console.log(productsUl);
+
+console.log(products);
+
+let ser = [];
+
+function renderProducts(products) {
+  products.forEach((product) => {
+    const li = document.createElement("li");
+
+    let shipsToHTML = "";
+    product.shipsTo.forEach(
+      (country) => (shipsToHTML += `<li>${country}</li>`)
+    );
+    li.innerHTML = "";
+
+    li.innerHTML = `
+    <ul>
+        <li>${product.name}</li>
+        <li>${product.price}</li>
+        <li>${product.rating}</li>
+        <ul class="ships-to">${shipsToHTML}</ul>
+    </ul>
+    `;
+    productsUl.appendChild(li);
+  });
```

Comment:
To filter both by text input and price input and countries shown, you want to filter the products list inside this function and only display those that satisfy user's conditions. 

So you'd have to modify your event listeners by 1) making the input values global variables (so that they're available to the `renderProducts()` and 2) removing the filter functions from the event listeners and having only _one_ filter function inside `renderProducts`. It would possibly take a few other minor modifications (you'll see in the process), but this is the gist of it.